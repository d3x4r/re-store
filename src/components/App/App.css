body {
  font-family: 'Playfair Display';
  font-weight: 400;
  font-size: 18px;
  line-height: 20px;

  color: #000000;
}
