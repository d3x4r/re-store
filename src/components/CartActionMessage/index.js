import CartActionMessage from './CartActionMessage';

export default CartActionMessage;
