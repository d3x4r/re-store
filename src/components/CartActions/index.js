import CartActions from './CartActions';

export default CartActions;
