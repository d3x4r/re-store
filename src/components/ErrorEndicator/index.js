import ErrorEndicator from './ErrorEndicator';

export default ErrorEndicator;
