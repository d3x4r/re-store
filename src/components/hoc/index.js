/* eslint-disable import/prefer-default-export */
import WithBookstoreService from './WithBookstoreService';

export { WithBookstoreService };
