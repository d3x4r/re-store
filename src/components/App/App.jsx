import React from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.css';
import Header from '../Header';
import { HomePage, CartPage } from '../pages';

const App = () => {
  return (
    <>
      <Header />
      <Switch>
        <div></div>
        <Route path="/cart" component={CartPage} />
        <Route path="/" component={HomePage} />
      </Switch>
    </>
  );
};

export default App;
