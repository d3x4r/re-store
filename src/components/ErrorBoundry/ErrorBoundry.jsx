/* eslint-disable react/no-unused-state */
/* eslint-disable react/state-in-constructor */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ErrorEndicator from '../ErrorEndicator';

class ErrorBoundry extends Component {
  state = {
    hasError: false,
  }

  componentDidCatch() {
    this.setState({ hasError: true });
  }

  render() {
    const { children } = this.props;
    const { hasError } = this.state;

    if (hasError) {
      return <ErrorEndicator />
    }

    return children;
  }
}

ErrorBoundry.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ErrorBoundry;
