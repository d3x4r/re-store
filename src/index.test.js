import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import App from './components/App';
import ErrorBoundry from './components/ErrorBoundry';
import BookstoreServive from './services';
import { BookstoreServiceProvider } from './components/bookstore-service-context';
import store from './store';

const bookstoreService = new BookstoreServive();

const AppContainer = () => (
    <Provider store={store}>
        <ErrorBoundry>
            <BookstoreServiceProvider value={bookstoreService}>
                <Router>
                    <App />
                </Router>
            </BookstoreServiceProvider>
        </ErrorBoundry>
    </Provider>
);

it('default render correctly', () => {
    const tree = renderer
        .create(<AppContainer />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});