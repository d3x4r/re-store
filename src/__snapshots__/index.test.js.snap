// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render correctly 1`] = `
Array [
  <header
    className="header"
  >
    <a
      className="header__logo-link"
      href="/"
      onClick={[Function]}
    >
      reStore
    </a>
    <div
      className="order-result"
    >
      <a
        aria-label="order"
        className="order-result__button btn btn-success btn-sm"
        href="/cart"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-shopping-cart"
        />
      </a>
      <span
        className="order-result__count"
      >
        The cart is empty
      </span>
      <span />
    </div>
  </header>,
  <div
    className="spinner"
  >
    <div
      className="loadingio-spinner-pulse-rbotfoul9n8"
    >
      <div
        className="ldio-pq0x2rh9c7q"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>,
]
`;

exports[`renders correctly 1`] = `
Array [
  <header
    className="header"
  >
    <a
      className="header__logo-link"
      href="/"
      onClick={[Function]}
    >
      reStore
    </a>
    <div
      className="order-result"
    >
      <a
        aria-label="order"
        className="order-result__button btn btn-success btn-sm"
        href="/cart"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-shopping-cart"
        />
      </a>
      <span
        className="order-result__count"
      >
        The cart is empty
      </span>
      <span />
    </div>
  </header>,
  <div
    className="spinner"
  >
    <div
      className="loadingio-spinner-pulse-rbotfoul9n8"
    >
      <div
        className="ldio-pq0x2rh9c7q"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>,
]
`;
